packages/winpanda/extra/src/winpanda/core/cmdconf.py

```diff
@@ -252,76 +253,173 @@ def get_dcos_conf(self):
                 )
 
-            for element in dcos_conf.get('package'):
-                if (not isinstance(element, dict) or not
-                        isinstance(element.get('path'), str) or not
-                        isinstance(element.get('content'), str)):
-                    raise cr_exc.RCElementError(
-                        f'DC/OS aggregated config: {element}'
-                    )
-
-            return dcos_conf
-
+            dstor_dcoscfg_pkg_path = lpi.get(dcos_conf_pkg_name).get(
+                'filename'
+            )
+            if not isinstance(dstor_dcoscfg_pkg_path, str):
+                raise cr_exc.RCElementError(
+                    f'DC/OS Linux package index: DC/OS config package'
+                    f' distribution storage path: {dstor_dcoscfg_pkg_path}'
+                )
         except cr_exc.RCError as e:
             raise e
         finally:
-            dcoscfg_fpath.unlink()
+            lpi_fpath.unlink()
+
+        dcoscfg_pkg_url = posixpath.join(
+            dstor_root_url, dstor_dcoscfg_pkg_path
+        )
+        dcoscfg_pkg_fname = Path(dstor_dcoscfg_pkg_path).name
+
+        # Download DC/OS aggregated configuration package ...
+        try:
+            cm_utl.download(dcoscfg_pkg_url, str(self.inst_storage.tmp_dpath))
+            LOG.debug(f'{self.msg_src}: DC/OS aggregated config: Download:'
+                      f' {dcoscfg_pkg_fname}: {dcoscfg_pkg_url}')
+        except Exception as e:
+            raise cr_exc.RCDownloadError(
+                f'DC/OS aggregated config: Download: {dcoscfg_pkg_fname}:'
+                f' {dcoscfg_pkg_url}: {type(e).__name__}: {e}'
+            ) from e
+
+        dcoscfg_pkg_fpath = self.inst_storage.tmp_dpath.joinpath(
+            dcoscfg_pkg_fname
+        )
+
+        try:
+            with tf.TemporaryDirectory(
+                dir=str(self.inst_storage.tmp_dpath)
+            ) as tmp_dpath:
+                cm_utl.unpack(str(dcoscfg_pkg_fpath), tmp_dpath)
+                LOG.debug(f'{self.msg_src}: DC/OS aggregated config: Extract:'
+                          f' OK')
+
+                values_fpath = Path(tmp_dpath).joinpath(values_fname)
+                values = cr_utl.rc_load_json(
+                    values_fpath,
+                    emheading=f'DC/OS aggregated config: Values: {values_fname}'
+                )
+                template_fpath = Path(tmp_dpath).joinpath(template_fname)
+                template = self.load_dcos_conf_templete(template_fpath)
+        except Exception as e:
+            if not isinstance(e, cr_exc.RCError):
+                raise cr_exc.RCExtractError(
+                    f'DC/OS aggregated config: {type(e).__name__}: {e}'
+                )
+            else:
+                raise
+        else:
+            return {'template': template, 'values': values}
+        finally:
+            dcoscfg_pkg_fpath.unlink()
+
+    @staticmethod
+    def load_dcos_conf_templete(fpath):
```

Comment:
The file is YAML. You should be able to replace this function with `yaml.load`. 

Also `templete` -> `template`

Reply:
Sure. It's YAML-formatted, but at the same time it's Jinja-template. holding Jinja formatted parts. And at the same time it incorporates multiple pices that are to become individual JSON-formatted configuration files after they are extracted from the dcos-config-windows.yaml. 
So to get the final *readable* JSON-configs from a source constructed this way we need to:
1) load dcos-config-windows.yaml by YAML-parser
2) process it as Jinja template by Jinja renderer
3) save individual parts as correct JSON-formatted files for JSON-parser to be able to parse them afterwards.

So we have a difficulty here:
1) when we do substitution for final JSON-content we ar to use double backslash for pathes for filese to be valid JSON, otherwise JSON parser would complain
2) when we do substitution for any other kind of content we generally should not do it because other parers (INI, YAML) do not expect double slashes and will possibly complain

This means we can't do substitution by Jinja, until we know which final format is aimed. I.e. before the we read the whole dcos-config-windows.yaml and identified all the individual parts and their desired formatting. So we need to load it first as a whole/ But at this stage besides being YAML formatted it's still a Jinja-template. And when we try to load it by yaml.safe_load() we got complains on Jinja placeholders containing curly braces. This is why we can't use YAML native parser.

Reply:
OK, the way I expected this to work was:

1. process the template file (using Jinja for now) - the template writer is responsible for dealing with JSON escapes
2. read the output as a YAML file
3. parse the YAML to create individual files

This requires a lot less guessing about the format required in the individual files.

Anyway, @DominikDary has requested that we merge the code to get a working demo, so I will note this as a change for GA.

Reply:
1) OK, for the GA plan.

.2) besides substitutions done with values from the expanded.config.full.json? brought along with the dcos-config-windows.yaml we are to support substitutions for local dynamic pathnames and winpanda have to decide it at runtime for which individual part from the dcos-config-windows.yaml to do what. I.e. use path variables with double backslashes or the same variables, but featuring single slashes. This is the main problem here. And fotr the moment I can't see any easy way how do we resolve it having this complicated design (multiformatted content inside a single plain text file) for configuration files delivery.

Reply:
what @jongiddy said for the ga frame.